generic/piranha-tree-sitter/src/cleanup_rules/java/rules.toml

```diff
@@ -0,0 +1,540 @@
+@ternary_expression)"""
+replace_node = "ternary_expression"
+replace = "@alternative"
+groups = ["if_cleanup"]
+
+# This rule is part (and entry point) for the inline local variable cleanup. Example: 
+# 
+# Before :
+# public foo someMethod() { 
+#  boolean isFlag = true; 
+#  somethingElse();
+# }
+#  
+# After :
+# public foo someMethod() { 
+#  somethingElse();
+# }
+#
+# Before inlining a variable we need to ensure that the variable is not re assigned anywhere within the method body 
+# to a value which is not the same as the initlaizer of variable we are inlining..
+# Note the below negative example of this rule where the variable `isFlag` is re-assigned to the value `getFlagStatus()` while it's 
+# initlaizer is `true`. In such a scenario the variable `isFlag` should not be inlined.
+# public foo someMethod() { 
+#  boolean isFlag = true; 
+#  isFlag = getFlagStatus();
+#  somethingElse();
+# }
+#
+
+[[rules]]
+name = "delete_variable_declaration"
+query = """
+(
+((local_variable_declaration 
+				declarator: (variable_declarator 
+                					name: (_) @variable_name
+                					value: [(true) (false)] @init)) @variable_declaration)
+)
+"""
+replace_node = "variable_declaration"
+replace = ""
+groups = []
+# Check if there is no assignment where the variable @variable_name is 
+# assigned to a value other than @init, within the method body
+# Please note that the tree-sitter queries in the constraint uses holes (i.e. `@variable_name` and `@init`).
+# These holes will be filled contextually based on the code snippet matched to `rule.query
+[[rules.constraints]]
+matcher = "(method_declaration) @md"
+queries = ["""
+(
+((assignment_expression
+    				left: (_) @a.lhs
+                    right: (_) @a.rhs) @a)
+(#eq? @a.lhs "@variable_name")
+(#not-eq? @a.rhs "@init")
+)"""]
+
+
+# This rule is part (and entry point) of the inline field declaration cleanup.
+# Same as the rule `delete_variable_declaration`, but applicable to 
+# fields. 
+[[rules]]
+name = "delete_field_declaration"
+query = """
+(
+((field_declaration 
+				declarator: (variable_declarator 
+                					name: (_) @variable_name
+                					value: [(true) (false) (parenthesized_expression (true)) (parenthesized_expression (false))] @init)) @field_declaration)
+)
+"""
+replace_node = "field_declaration"
+replace = ""
+groups = []
+# Check if there is no assignment where the variable @variable_name is 
+# assigned to a value other than @init, within the method body
+[[rules.constraints]]
+matcher = "(class_declaration) @cd"
+queries = ["""(
+((assignment_expression
+    				left: (_) @a.lhs
+                    right: (_) @a.rhs) @a)
+(#eq? @a.lhs "@variable_name")
+(#not-eq? @a.rhs "@init")
+)"""]
+
+
+# This rule is part of inline variable declaration and field declaration cleanup. 
+# This rule is applied after the above two rule - `delete_field_declaration` and `delete_variable_declaration`.
```

Comment:
Maybe get rid of this comment as the edges.toml contains the spec of how this will be applied. This is to ensure that this comment doesn't become stale with time (even though it is accurate now)